contracts/WitnetRequestsBoard.sol

```diff
@@ -365,8 +380,17 @@ contract WitnetRequestsBoard is WitnetRequestsBoardInterface {
       _epoch,
       _index,
       resHash), "Invalid PoI");
-    address payable relayer = payable(blockRelay.readRelayerAddress(_blockHash, _epoch));
-    relayer.transfer(requests[_id].blockReward);
+
+    // Transfer the relayer payment to the block relayer or the requestor in case that the relayer has already been paid
```

Comment:
It is just a suggestion, subject to be ignored without hard feelings! :smile: 
```suggestion
    // Transfer the block reward to the relayer only if it has not already been paid, otherwise return it to the requestor
```